janitor/functions/complete.py

```diff
@@ -212,112 +212,141 @@ def _computations_complete(
     # trade-off for the simplicity of merge is not so bad
     # of course there could be a better way ...
     if by is None:
-        uniques = _generic_complete(df, columns, all_strings)
+        uniques = _generic_complete(df, columns, all_strings, sort)
     else:
         uniques = df.groupby(by)
-        uniques = uniques.apply(_generic_complete, columns, all_strings)
+        uniques = uniques.apply(_generic_complete, columns, all_strings, sort)
         uniques = uniques.droplevel(-1)
         column_checker = by + column_checker
-    if fill_value is None:
-        return df.merge(
-            uniques, on=column_checker, how="outer", sort=sort, copy=False
-        )
-    # if fill_value is present
-    if is_scalar(fill_value):
-        # faster when fillna operates on a Series basis
-        fill_value = {col: fill_value for col in df}
-
-    if explicit:
-        return df.merge(
-            uniques, on=column_checker, how="outer", sort=sort, copy=False
-        ).fillna(fill_value, downcast="infer")
-    # keep only columns that are not part of column_checker
-    # IOW, we are excluding columns that were not used
-    # to generate the combinations
-    fill_value = {
-        col: value
-        for col, value in fill_value.items()
-        if col not in column_checker
-    }
-    if not fill_value:  # there is nothing to fill
-        return df.merge(
-            uniques, on=column_checker, how="outer", sort=sort, copy=False
-        )
-
-    # when explicit is False
-    # filter out rows from `unique` that already exist in the parent dataframe
-    # fill the null values in the trimmed `unique`
-    # and merge back to the main dataframe
 
-    # to get a name that does not exist in the columns
-    indicator = "".join(df.columns)
-    trimmed = df.loc(axis=1)[column_checker]
-    uniques = uniques.merge(
-        trimmed, how="left", sort=False, copy=False, indicator=indicator
+    columns = df.columns
+    indicator = False
+    if (fill_value is not None) and (explicit is False):
```

Comment:
Since explicit is already a boolean, we can just do `not explicit` instead.

```suggestion
    if fill_value is not None and not explicit:
```